core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/DeprecatedConfigMojo.java

```diff
@@ -47,6 +48,18 @@
     @Parameter( defaultValue = "${basedir}", readonly = true )
     private File basedir;
 
+    /**
+     * @deprecated bean read only
+     */
+    @Parameter(  defaultValue = "${project.artifact}", readonly = true )
+    private Artifact deprecatedBeanReadOnly;
+
+    /**
+     * @deprecated bean
+     */
+    @Parameter(  defaultValue = "${project.artifact}" )
+    private Artifact deprecatedBean;
```

Comment:
I don't see the corresponding test...

Reply:
It is corner case when internal value, project object is used as property without read only flags.
Such parameter should not generated warning.

Reply:
I see, there is one space too much